ginga/rv/plugins/Header.py

```diff
@@ -97,39 +102,23 @@ def build_gui(self, container):
         btn.add_callback('activated', lambda w: self.help())
         btns.add_widget(btn, stretch=0)
         btns.add_widget(Widgets.Label(''), stretch=1)
-        vbox.add_widget(btns, stretch=0)
+        top.add_widget(btns, stretch=0)
 
-        container.add_widget(vbox, stretch=1)
+        container.add_widget(top, stretch=1)
         self.gui_up = True
 
-    def _create_header_window(self, info):
-        vbox = Widgets.VBox()
-        vbox.set_margins(2, 2, 2, 2)
-
-        color_alternate = self.settings.get('color_alternate_rows', True)
-        table = Widgets.TreeView(auto_expand=True,
-                                 use_alt_row_color=color_alternate)
-        self.table = table
-        table.setup_table(self.columns, 1, 'key')
-
-        vbox.add_widget(table, stretch=1)
-
-        info.setvals(widget=vbox, table=table)
-        return vbox
-
-    def set_header(self, info, image):
-        if self._image == image:
+    def set_header(self, image):
```

Comment:
Well, to be fair, most plugin methods are not designed to be callable outside the plugin, and that has been a more recent development.  In the case of this method, the user would have to construct an `info` data structure compatible with what is built internally in the plugin, and then it would be showing a header for an image that is not visible in the viewer.

Shall I prefix the method with an underscore?

Reply:
My concern is indeed more of people subclassing, say, this plugin in their custom plugin to reduce duplicated code. Come to think of it, perhaps I am the only one that lazy. As I am not subclassing this particular plugin right now, I am ok with keeping these changes. I do worry about future API change like this in a plugin that I subclass (e.g., `Mosaic`).

As for deprecation policy, I have seen different packages handle it differently and not seen any "one right way" to do this. Given 3.0 is backward incompatible in regards to PY2, maybe it won't be so bad to have this in too without deprecation. Is adding deprecation period more pain than it's worth for such a case? 🤔

Reply:
Hmm, I think so for this case at least.  It gets rid of a big messy data structure that doesn't need to be passed around anymore.  If someone passed something in it wouldn't know what to do with it!

I think maybe we can catagorize this as a 3.0 major release type of change.

But good to think about for the future.  Maybe we should have a review of plugins after 3.0 to see if there are some methods that should be changed to private.